### PR TITLE
Network

### PR DESCRIPTION
**Pull request by mel-splunk**
Review complete.  No DNS as waiting for Quentin to write monitor for DNS, so we have metric names

### CODE REVIEW

File: network/http/webcheck/detectors-webcheck.tf [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+resource "signalfx_detector" "invalid_tls_certificate" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] TLS invalid certificate count"
+
+	program_text = <<-EOF
+		signal = data('http.cert_valid', ${module.filter-tags.filter_custom}).count()${var.invalid_tls_certificate_aggregation_function}.${var.invalid_tls_certificate_transformation_function}(over='${var.invalid_tls_certificate_transformation_window}').publish('signal')
+		detect(when(signal < ${var.invalid_tls_certificate_threshold_critical})).publish('CRIT')
+	EOF
+
+	rule {
+		description           = " < ${var.invalid_tls_certificate_threshold_critical}"
```
Comment (xp-1000) [inline network/http/webcheck/detectors-webcheck.tf:217]: hardcode value and remove variable using a message like `TLS Certificate is not valid`

File: network/http/webcheck/detectors-webcheck.tf [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	rule {
+		description           = "is too low < ${var.http_code_matched_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.http_code_matched_disabled_critical, var.http_code_matched_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.http_code_matched_notifications_critical, var.http_code_matched_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:38]: removed

File: network/http/webcheck/detectors-webcheck.tf [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] HTTP code matched"
+
+	program_text = <<-EOF
+		signal = data('http.code_matched', ${module.filter-tags.filter_custom}).count()${var.http_code_matched_aggregation_function}.${var.http_code_matched_transformation_function}(over='${var.http_code_matched_transformation_window}').publish('signal')
+		detect(when(signal < ${var.http_code_matched_threshold_critical})).publish('CRIT')
+		detect(when(signal < ${var.http_code_matched_threshold_warning}) and when(signal >= ${var.http_code_matched_threshold_critical})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too low < ${var.http_code_matched_threshold_critical}"
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:30]: updated

File: network/http/webcheck/detectors-webcheck.tf [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "http_code_matched" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] HTTP code matched"
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:21]: updated

File: network/http/webcheck/variables.tf [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+variable "http_content_length_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of http_content_length detector"
+  type        = list
+  default     = []
+}
+
+variable "http_content_length_aggregation_function" {
+  description = "Aggregation function and group by for http_content_length detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/variables.tf:354]: update mean by url

File: network/http/webcheck/variables.tf [hunk 6]
```diff
@@ -0,0 +1,571 @@
...
+variable "certificate_expiration_date_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of certificate_expiration_date detector"
+  type        = list
+  default     = []
+}
+
+variable "certificate_expiration_date_aggregation_function" {
+  description = "Aggregation function and group by for certificate_expiration_date detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/variables.tf:422]: updated

File: network/http/webcheck/detectors-webcheck.tf [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+		notifications         = coalescelist(var.tls_certificate_expiration_notifications_warning, var.tls_certificate_expiration_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "invalid_tls_certificate" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] TLS invalid certificate count"
+
+	program_text = <<-EOF
+		signal = data('http.cert_valid', ${module.filter-tags.filter_custom}).count()${var.invalid_tls_certificate_aggregation_function}.${var.invalid_tls_certificate_transformation_function}(over='${var.invalid_tls_certificate_transformation_window}').publish('signal')
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:212]: updated

File: network/http/webcheck/detectors-webcheck.tf (hunk 1 above)
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:217]: updated

File: network/http/webcheck/detectors-webcheck.tf [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+		description           = " < ${var.certificate_expiration_date_threshold_warning} days"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.certificate_expiration_date_disabled_warning, var.certificate_expiration_date_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.certificate_expiration_date_notifications_warning, var.certificate_expiration_date_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "tls_certificate_expiration" {
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:180]: removed

File: network/http/webcheck/detectors-webcheck.tf [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "http_code_matched" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] HTTP code matched"
+
+	program_text = <<-EOF
+		signal = data('http.code_matched', ${module.filter-tags.filter_custom}).count()${var.http_code_matched_aggregation_function}.${var.http_code_matched_transformation_function}(over='${var.http_code_matched_transformation_window}').publish('signal')
```
[the remark this answers is not in the document]
Comment (ybjosephson) [reply network/http/webcheck/detectors-webcheck.tf:24]: updated

File: network/http/webcheck/detectors-webcheck.tf [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+		detect(when(signal < ${var.http_code_matched_threshold_critical})).publish('CRIT')
```
[the remark this answers is not in the document]
Comment (pdecat) [reply network/http/webcheck/detectors-webcheck.tf:25]: Also, this errors:
```
Error: Bad status 400: {
  "code" : 400,
  "message" : "Error with program: [File \"<program>\", line 2, in \ndetect(when(signal < 1)).publish('CRIT')\nSyntaxError: invalid syntax]"
}

  on .terraform/modules/signalfx.signalfx-detectors-network-http-webcheck/network/http/webcheck/detectors-webcheck.tf line 20, in resource "signalfx_detector" "http_code_matched":
  20: resource "signalfx_detector" "http_code_matched" {
```

File: network/http/webcheck/detectors-webcheck.tf [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+}
+
+resource "signalfx_detector" "http_regex_matched" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] HTTP regex matched"
+
+	program_text = <<-EOF
+		signal = data('http.regex_matched', ${module.filter-tags.filter_custom}).count()${var.http_regex_matched_aggregation_function}.${var.http_regex_matched_transformation_function}(over='${var.http_regex_matched_transformation_window}').publish('signal')
+		detect(when(signal < ${var.http_regex_matched_threshold_critical})).publish('CRIT')
```
[the remark this answers is not in the document]
Comment (pdecat) [reply network/http/webcheck/detectors-webcheck.tf:44]: Also, this errors:

```
Error: Bad status 400: {
  "code" : 400,
  "message" : "Error with program: [File \"<program>\", line 2, in \ndetect(when(signal < 1)).publish('CRIT')\nSyntaxError: invalid syntax]"
}

  on .terraform/modules/signalfx.signalfx-detectors-network-http-webcheck/network/http/webcheck/detectors-webcheck.tf line 39, in resource "signalfx_detector" "http_regex_matched":
  39: resource "signalfx_detector" "http_regex_matched" {
```

File: network/http/webcheck/detectors-webcheck.tf (hunk 10 above)
[the remark this answers is not in the document]
Comment (pdecat) [reply network/http/webcheck/detectors-webcheck.tf:25]: Error resolved with last update.

File: network/http/webcheck/detectors-webcheck.tf (hunk 11 above)
[the remark this answers is not in the document]
Comment (pdecat) [reply network/http/webcheck/detectors-webcheck.tf:44]: Error resolved with last update.

**Pull request merged**